packages/react/src/xcss/xcss-prop.ts

```diff
@@ -0,0 +1,50 @@
+import type * as CSS from 'csstype';
+
+import type { Pseudos as CSSPseudos } from './pseudos';
+
+export type CSSRuleDefinition = CSSProperties & CSSPseudoRule;
+
+type CSSProperties = Readonly<CSS.PropertiesFallback<number | string>>;
+
+type CSSPseudoRule = { [Q in CSSPseudos]?: CSSProperties };
+
+declare const __classref: unique symbol;
+export type CompiledStyleClassReference = { [__classref]: true };
+
+type XCSSItem<TStyleDecl extends keyof CSSProperties> = {
+  [Q in keyof CSSProperties]: Q extends TStyleDecl ? CompiledStyleClassReference : never;
+};
+
+type XCSSPseudos<K extends keyof CSSProperties, TPseudos extends CSSPseudos> = {
+  [Q in CSSPseudos]?: Q extends TPseudos ? XCSSItem<K> : never;
+};
+
+/**
+ * __xcss prop__
+ *
+ * Controls the type of `xcss` supplied to a component allowing for
+ * more restrictive typing.
+ *
+ * @example
+ * ```tsx
+ * type Props = {
+ *   xcss: XCSSProp<'color', '&:hover'>
+ * }
+ *
+ * const Component = ({ xcss }: Props) => {
+ *
+ * }
+ *
+ * // Valid usage
+ * <Component xcss={{ color: 'red', '&:hover': { color: 'blue' } }} />
```

Comment:
This isn't actually correct they need to be flushed through `css` or equivalent.

Reply:
Yep good shout.